### ISSUE TITLE
Wrong  unsupported-operator error

### ISSUE THREAD

**Issue by pmareke** [astral-sh/ty#540]
### Summary

Hi team, first of all thanks for ty, it's an amazing tool, I'm using a lot on my personal projects.

In one of them I have the following error:

```sh
error[unsupported-operator]: Operator `<` is not supported for types `float` and `None`, in comparing `float` with `int | float | None`
```

With the following code:

```python
@dataclass
class MyClass:
    min: float | None = None


def lower_value(my_class: MyClass, value: str) -> bool:
    if my_class.min:
        return float(value) < my_class.min
    return False
```

Same with this code:

```python
@dataclass
class MyClass:
    min: float | None = None


def lower_value(my_class: MyClass, value: str) -> bool:
    if my_class.min is not None:
        return float(value) < my_class.min
    return False
```

And I think the code is correct (this code in `mypy` doesn't raise any error).

Thanks and sorry if this is not a bug or it's repeated.

### Version

0.0.1a7

**Comment by jelle-openai** [comment]
This is an instance of #164.

**Comment by pmareke** [comment]
> This is an instance of [#164](https://github.com/astral-sh/ty/issues/164).

Thanks and sorry @jelle-openai.

I searched for "unsupported-operator" and I didn't see anything similar.
